packages/discord.js-utilities/src/lib/MessagePrompter.ts

```diff
@@ -0,0 +1,402 @@
+import type {
+	Message,
+	User,
+	TextChannel,
+	NewsChannel,
+	APIMessageContentResolvable,
+	MessageOptions,
+	MessageAdditions,
+	DMChannel,
+	CollectorFilter,
+	EmojiIdentifierResolvable,
+	MessageReaction,
+	EmojiResolvable,
+	GuildEmoji,
+	ReactionEmoji
+} from 'discord.js';
+
+/**
+ * This is a [[MessagePrompter]], a utility that sends a prompt message that can resolve any kind of input.
+ * There are several specifiable types to prompt for user input, and they are as follows:
+ * - Confirm
+ *   This will send a simple Yes/No prompt, using reactions.
+ * - Number
+ *   This will prompt for an integer. You can specify a range, however by default it will use 0 - 10.
+ * - Reactions
+ *   This can be any kind of reaction emoji Discord supports and as many as you want. This type will return that reaction instead of a boolean.
+ * - Message
+ *   This will prompt the user and require a response in the form of a message. This can be helpful if you require a user to upload an image for example, or give text input.
+ *
+ * You must either use this class directly or extend it.
+ *
+ * [[MessagePrompter]] uses reactions to prompt for a yes/no answer and returns it.
+ * You can modify the confirm and cancel reaction used for each message, or use the [[MessagePrompter.defaultPrompts]].
+ * [[MessagePrompter.defaultPrompts]] is also static so you can modify these directly.
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Are you sure you want to continue?');
+ * const confirmed = await handler.run(channel, author);
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Choose a number between 5 and 10?', {
+ * 		type: 'number',
+ * 		reactions: {
+ * 			start: 5,
+ * 			end: 10
+ * 		}
+ * });
+ * const confirmed = await handler.run(channel, author);
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Are you happy or sad?', {
+ * 		type: 'reaction',
+ * 		reactions: ['🙂', '🙁']
+ * });
+ * const confirmed = await handler.run(channel, author);
```

Comment:
`confirmed`? This could use a better name.

Reply:
Suggestion?